examples/albert/run_trainer.py

```diff
@@ -0,0 +1,317 @@
+    logger.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
+        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
+    )
+    # Set the verbosity to info of the Transformers logger (on main process only):
+    if is_main_process(training_args.local_rank):
+        transformers.utils.logging.set_verbosity_info()
+        transformers.utils.logging.enable_default_handler()
+        transformers.utils.logging.enable_explicit_format()
+    logger.info("Training/evaluation parameters %s", training_args)
+
+
+def get_model(training_args, config, tokenizer):
+    # Find latest checkpoint in output_dir
+    output_dir = Path(training_args.output_dir)
+    logger.info(f'Checkpoint dir {output_dir}, contents {list(output_dir.glob("checkpoint*"))}')
+    latest_checkpoint_dir = max(output_dir.glob('checkpoint*'), default=None, key=os.path.getctime)
+
+    if latest_checkpoint_dir is not None:
+        logger.info(f'Loading model from {latest_checkpoint_dir}')
+        model = AlbertForPreTraining.from_pretrained(latest_checkpoint_dir)
+    else:
+        logger.info(f'Training from scratch')
+        model = AlbertForPreTraining(config)
+        model.resize_token_embeddings(len(tokenizer))
+
+    return model
+
+
+def get_optimizer_and_scheduler(training_args, model):
+    no_decay = ["bias", "LayerNorm.weight"]
+    optimizer_grouped_parameters = [
+        {
+            "params": [p for n, p in model.named_parameters() if not any(nd in n for nd in no_decay)],
+            "weight_decay": training_args.weight_decay,
+        },
+        {
+            "params": [p for n, p in model.named_parameters() if any(nd in n for nd in no_decay)],
+            "weight_decay": 0.0,
+        },
+    ]
+
+    opt = Lamb(
+        optimizer_grouped_parameters,
+        lr=training_args.learning_rate,
+        betas=(training_args.adam_beta1, training_args.adam_beta2),
+        eps=training_args.adam_epsilon,
+        weight_decay=training_args.weight_decay,
+        clamp_value=training_args.clamp_value,
+        debias=True,
+    )
+
+    scheduler = get_linear_schedule_with_warmup(
+        opt,
+        num_warmup_steps=training_args.warmup_steps,
+        num_training_steps=training_args.max_steps
+    )
+
+    return opt, scheduler
+
+
+class CollaborativeCallback(transformers.TrainerCallback):
+    def __init__(self, dht: hivemind.DHT, collaborative_optimizer: hivemind.CollaborativeOptimizer,
+                 model: torch.nn.Module, trainer_uuid: str, statistics_expiration: float):
+        self.dht, self.collaborative_optimizer = dht, collaborative_optimizer
+        self.model = model
+        self.trainer_uuid, self.statistics_expiration = trainer_uuid, statistics_expiration
+        self.last_reported_collaboration_step = -1
+        self.previous_state = self.get_current_state()
+        super().__init__()
```

Comment:
It's best to move `__init__` of a parent class to the top of this method

Reply:
thanks, did that

Reply:
I meant that we can run super().__init__() first and initialize all other fields later

Reply:
sry, i forgot push that change to origin